drivers/SmartThings/matter-appliance/src/matter-cook-top/init.lua

```diff
@@ -0,0 +1,233 @@
+  end
+end
+
+local function is_cook_top_device(opts, driver, device, ...)
+  local cook_top_eps = get_endpoints_for_dt(device, COOK_TOP_DEVICE_TYPE_ID)
+  local oven_eps = get_endpoints_for_dt(device, OVEN_DEVICE_ID)
+  -- We want to skip lifecycle events in cases where device is an oven with a composed cook-top device
+  if (#oven_eps > 0) and opts.dispatcher_class == "DeviceLifecycleDispatcher" then
+    return false
+  end
+  if #cook_top_eps > 0 then
+    return true
+  end
+  return false
+end
+
+-- Matter Handlers --
+local function selected_temperature_level_attr_handler(driver, device, ib, response)
+  local tl_eps = embedded_cluster_utils.get_endpoints(device, clusters.TemperatureControl.ID,
+    { feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_LEVEL })
+  if #tl_eps == 0 then
+    log.warn_with({hub_logs = true}, string.format("Device does not support TEMPERATURE_LEVEL feature"))
+    return
+  end
+  log.info(string.format("selected_temperature_level_attr_handler: %s", ib.data.value))
+
+  local temperatureLevel = ib.data.value
+  local supportedTemperatureLevelsMap = device:get_field(SUPPORTED_TEMPERATURE_LEVELS_MAP) or {}
+  local supportedTemperatureLevels = supportedTemperatureLevelsMap[ib.endpoint_id] or {}
+  if supportedTemperatureLevels[temperatureLevel+1] then
+    local tempLevel = supportedTemperatureLevels[temperatureLevel+1]
+    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.temperatureLevel.temperatureLevel(tempLevel))
+  end
+  log.warn("Received unsupported temperature level for endpoint "..(ib.endpoint_id))
+end
+
+local function supported_temperature_levels_attr_handler(driver, device, ib, response)
+  local tl_eps = embedded_cluster_utils.get_endpoints(device, clusters.TemperatureControl.ID,
+    { feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_LEVEL })
+  if #tl_eps == 0 then
+    log.warn_with({hub_logs = true}, string.format("Device does not support TEMPERATURE_LEVEL feature"))
+    return
+  end
+
+  local supportedTemperatureLevelsMap = device:get_field(SUPPORTED_TEMPERATURE_LEVELS_MAP) or {}
+  local supportedTemperatureLevels = {}
+  for _, tempLevel in ipairs(ib.data.elements) do
+    log.info(string.format("supported_temperature_levels_attr_handler: %s", tempLevel.value))
+    table.insert(supportedTemperatureLevels, tempLevel.value)
+  end
+  supportedTemperatureLevelsMap[ib.endpoint_id] = supportedTemperatureLevels
+  device:set_field(SUPPORTED_TEMPERATURE_LEVELS_MAP, supportedTemperatureLevelsMap, { persist = true })
+  local event = capabilities.temperatureLevel.supportedTemperatureLevels(supportedTemperatureLevels,
+    { visibility = { displayed = false } })
+  device:emit_event_for_endpoint(ib.endpoint_id, event)
+end
+
+local function temp_event_handler(driver, device, ib, response)
+  device.log.info(string.format("temp_event_handler: %s", ib.data.value))
+
+  local temp
+  local unit = "C"

```

Comment:
So we can only handle `temperatureMeasurement` in Celsius? Is this acceptable?

Reply:
Good question! The measured value from the TemperatureMeasurement cluster will always be in degC because it has the temperature data type: https://librarian-regionals.smartthingsgdev.com/matter/Matter_1-3/index.html#ref_DataTypeTemperature

Therefore, this value will always be degC from the driver perspective because this is handling the cluster attribute value, which is always in C. 

The other temp conversion consideration you may be thinking of is due to receiving a capability command, not a cluster attribute. The capability command does not have a clear unit associated with the value, which is where we run into the issues.

So I think this is appropriate for the to use degC to emit the capability event because that is the unit of the value. The hubcore temperature conversion should then localize the unit to degF if needed before sending the value to the cloud to be displayed to the user.

CC: @nickolas-deboom